tests/Dockerfile

```
@@ -0,0 +1,60 @@
+FROM ubuntu:rolling
+LABEL org.opencontainers.image.source https://github.com/elastisys/compliantkubernetes-apps
+
+ARG DEBIAN_FRONTEND=noninteractive
+ARG TZ=Etc/UTC
+
+RUN apt-get update && \
+    apt-get install -y apache2-utils bats curl dnsutils gettext-base git gpg iputils-ping jq make net-tools npm pwgen s3cmd ssh unzip wget \
+        libasound2 libgbm-dev libgconf-2-4 libgtk2.0-0 libgtk-3-0 libnotify-dev libnss3 libxss1 libxtst6 xauth xvfb && \
+    apt-get clean && \
+    rm -rf /var/lib/apt/lists/*
+
+ENV SOPS_VERSION="3.7.3"
+RUN wget --progress=dot:giga "https://github.com/mozilla/sops/releases/download/v${SOPS_VERSION}/sops-v${SOPS_VERSION}.linux" && \
+    install -Tm 755 "sops-v${SOPS_VERSION}.linux" /usr/local/bin/sops && \
+    rm "sops-v${SOPS_VERSION}.linux"
+
+ENV KUBECTL_VERSION "v1.24.4"
+RUN wget --progress=dot:giga "https://storage.googleapis.com/kubernetes-release/release/${KUBECTL_VERSION}/bin/linux/amd64/kubectl" && \
+    install -Tm 755 kubectl /usr/local/bin/kubectl && \
+    rm kubectl
+
+ENV KUBELOGIN_VERSION="v1.24.0"
+RUN wget --progress=dot:giga "https://github.com/int128/kubelogin/releases/download/${KUBELOGIN_VERSION}/kubelogin_linux_amd64.zip" && \
+    unzip kubelogin_linux_amd64.zip && \
+    install -Tm 755 kubelogin /usr/local/bin/kubectl-oidc_login && \
+    rm kubelogin kubelogin_linux_amd64.zip LICENSE README.md
+
+ENV HELM_VERSION "v3.8.0"
+RUN wget --progress=dot:giga "https://get.helm.sh/helm-${HELM_VERSION}-linux-amd64.tar.gz" && \
+    tar -zxvf "helm-${HELM_VERSION}-linux-amd64.tar.gz" && \
+    install -Tm 755 linux-amd64/helm /usr/local/bin/helm && \
+    rm -rf linux-amd64 "helm-${HELM_VERSION}-linux-amd64.tar.gz"
+
+ENV XDG_DATA_HOME=/root/.config
+RUN helm plugin install https://github.com/databus23/helm-diff --version v3.5.0
+RUN helm plugin install https://github.com/jkroepke/helm-secrets --version v3.12.0
+
+ENV HELMFILE_VERSION "0.154.0"
+RUN wget --progress=dot:giga "https://github.com/helmfile/helmfile/releases/download/v${HELMFILE_VERSION}/helmfile_${HELMFILE_VERSION}_linux_amd64.tar.gz" && \
+    tar -zxvf "helmfile_${HELMFILE_VERSION}_linux_amd64.tar.gz" helmfile && \
+    install -Tm 755 helmfile /usr/local/bin/helmfile && \
+    rm helmfile "helmfile_${HELMFILE_VERSION}_linux_amd64.tar.gz"
+
+ENV YQ_VERSION "v4.26.1"
+RUN wget --progress=dot:giga "https://github.com/mikefarah/yq/releases/download/${YQ_VERSION}/yq_linux_amd64" && \
+    install -Tm 755 yq_linux_amd64 /usr/local/bin/yq4 && \
+    rm yq_linux_amd64
+
+COPY ./ /tests
+
+RUN chown -R ubuntu:ubuntu /tests
+
+WORKDIR /tests
+
+USER ubuntu
+
+RUN ls -l
+
+RUN make dep
```

Would it be possible to only copy the necessary files for `make dep` and then the rest to reduce build time when updating tests?

---

I just tried this and it _seems_ to be working. The update test -> build -> run round trip time is reduced a lot by this so it would be nice to see something to this effect.

```suggestion
COPY ./Makefile /tests/Makefile
COPY ./common /tests/common

RUN chown -R ubuntu:ubuntu /tests

WORKDIR /tests

USER ubuntu

RUN ls -l

RUN make dep

COPY ./ /tests
```

---

Yes, I was going to start fixing that. :+1:

---

Also fixed so it generates tests during build.

---

Sweet!